components/layout_2020/flow/mod.rs

```diff
@@ -1013,19 +1034,35 @@ impl NonReplacedFormattingContext {
                         block_size: block_size.map(|t| t.into()),
                         style: &self.style,
                     },
+                    containing_block,
                 );
-                content_size = LogicalVec2 {
-                    inline: proposed_inline_size,
-                    block: block_size.auto_is(|| {
-                        layout
-                            .content_block_size
-                            .clamp_between_extremums(
-                                min_box_size.block.into(),
-                                max_box_size.block.map(|t| t.into()),
-                            )
-                            .into()
-                    }),
-                };
+
+                if let Some(inline_size) = layout.content_inline_size_for_table {
+                    let outer_inline_size = inline_size + pbm.padding_border_sums.inline;
+                    placement.set_inline_size(outer_inline_size, &pbm);
+                    if outer_inline_size > placement_rect.size.inline {
+                        positioning_context.truncate(&positioning_context_length);
+                        continue;
+                    }
+                    content_size = LogicalVec2 {
+                        block: layout.content_block_size,
+                        inline: inline_size,
+                    }
+                    .into();
+                } else {
+                    content_size = LogicalVec2 {
+                        block: block_size.auto_is(|| {
+                            layout
+                                .content_block_size
+                                .clamp_between_extremums(
+                                    min_box_size.block.into(),
+                                    max_box_size.block.map(|t| t.into()),
+                                )
+                                .into()
```

Comment:
Can this be:

```suggestion
                                .content_block_size
                                .into()
                                .clamp_between_extremums(
                                    min_box_size.block,
                                    max_box_size.block,
                                )
```

Reply:
Nice! But "type annotations needed", so

```suggestion
                            Length::from(layout.content_block_size)
                                .clamp_between_extremums(min_box_size.block, max_box_size.block)
```